File: apps/web/src/shared/api/health.ts
import { apiClient } from '@/shared/api/client'

export type HealthResponse = {
  status: string
  service: string
  version: string
  environment: string
  currency: string
  locale: string
  timestamp: string
}

export async function fetchHealth(): Promise<HealthResponse> {
  const { data } = await apiClient.get<HealthResponse>('/health')
  return data
}

File: apps/api/app/api/v1/endpoints/health.py
from datetime import UTC, datetime

from fastapi import APIRouter
from pydantic import BaseModel, Field

from app.core.config import get_settings

router = APIRouter()


class HealthResponse(BaseModel):
    status: str = Field(examples=["ok"])
    service: str
    version: str
    environment: str
    currency: str
    locale: str
    timestamp: datetime


@router.get("/health", response_model=HealthResponse)
def health_check() -> HealthResponse:
    settings = get_settings()
    return HealthResponse(
        status="ok",
        service=settings.app_name,
        version=settings.app_version,
        environment=settings.app_env,
        currency=settings.currency_code,
        locale=settings.default_locale,
        timestamp=datetime.now(UTC),
    )
